app/src/main/java/com/example/androidtraining/ui/product/ProductScreen.kt

```diff
@@ -0,0 +1,237 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.paint
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil.compose.rememberAsyncImagePainter
+import com.example.androidtraining.R
+import com.example.androidtraining.ui.theme.Green
+import com.example.androidtraining.ui.theme.Purple
+import com.example.androidtraining.ui.theme.White
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ProductScreen() {
+    val viewModel: ProductViewModel = hiltViewModel()
+    val context = LocalContext.current
+    val successfulProductLoading by viewModel.successfulProductLoading.collectAsState()
+    val paddingLarge = dimensionResource(id = R.dimen.padding_large)
+    val paddingMedium = dimensionResource(id = R.dimen.padding_medium)
+    LaunchedEffect(true) {
+        viewModel.getProduct()
+    }
+
+    LaunchedEffect(successfulProductLoading) {
+        successfulProductLoading?.let {
+            if (!it) Toast.makeText(context, R.string.err_product_loading, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    Scaffold(
+        topBar = {
+            CenterAlignedTopAppBar(
+                title = {
+                    Text(
+                        stringResource(id = R.string.item),
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis
+                    )
+                },
+                navigationIcon = {
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(
+                            imageVector = Icons.Outlined.ArrowBack,
+                            contentDescription = stringResource(id = R.string.arrow_back)
+                        )
+                    }
+                },
+                actions = {
+                    IconButton(onClick = { /* doSomething() */ }) {
```

Comment:
Optional as it's not such a big layout but can be exported to a composable of its own.